### PR TITLE
Allow Villagers to farm more crops

### PR DESCRIPTION
**Pull request by temp1011**
Motivation: https://github.com/Ellpeck/ActuallyAdditions/issues/959

At the moment the villager AI for harvesting and replanting crops is hardcoded. This PR adds to this AI for all crops that have `EnumPlantType.Crop`.

Any tips for the patch would be helpful as it doesn't look nice at all at the moment. Also any issues (Maybe two high crops like IE Hemp may cause issues?).

### CODE REVIEW

**Comment by josephcsible** [comment]
What's with the two unrelated commits?

File: patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+                     {
++                        net.minecraft.item.Item item = itemstack.func_77973_b();
+                         if (itemstack.func_77973_b() == Items.field_151014_N)
+                         {
+                             world.func_180501_a(blockpos, Blocks.field_150464_aj.func_176223_P(), 3);
+@@ -94,6 +95,7 @@
+                             world.func_180501_a(blockpos, Blocks.field_185773_cZ.func_176223_P(), 3);
+                             flag = true;
+                         }
++                        else if (item instanceof net.minecraftforge.common.IPlantable && ((net.minecraftforge.common.IPlantable)item).getPlantType(world, blockpos)==net.minecraftforge.common.EnumPlantType.Crop) { world.func_180501_a(blockpos, ((net.minecraftforge.common.IPlantable)item).getPlant(world, blockpos), 3); flag=true;}
```
Comment (Gegy) [inline patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch:15]: This can be put over multiple lines,

**Comment by jabelar** [comment]
The villager farming isn't really "hard coded". The task list for villagers (like all entities) is public and modifiable already. So you can simply replace the AI with one that does what you want. Basically I don't think that this PR is really that necessary.

**Comment by Daomephsta** [comment]
@jabelar 
Yes, that would be a functional solution. However it would also be a fragile solution.  
There are a lot of mods that add crops, if they're all going to replace that AI task they're going to have to do it in a compatible way. That means somehow marking their modified variant as the same task (Probably by extending the vanilla villager farming task class) and wrapping whatever farming task they find so that it continues to run its code whether it's vanilla or not.  
If a mod forgets to mark their modified variant as the same task, they break the wrapping chain for every mod that attempts to wrap the task after them. If a mod forgets to wrap or doesn't wrap properly they break the wrapping chain for every mod that wrapped the task before them.

I think this PR is very much needed as it is much more robust than the existing solution.

File: patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch (hunk 1 above)
Comment (temp1011) [reply patches/minecraft/net/minecraft/entity/ai/EntityAIHarvestFarmland.java.patch:15]: Will do.

**Comment by temp1011** [comment]
My branch seems pretty broken. I'll try and fix but looks like I need a new PR.

**Pull request closed without merging**